### PR TITLE
Combine arm and elevator subsystems

### LINKED ISSUE #61: Auton does not run after robot is enabled in teleop

### PR DESCRIPTION
**Pull request by citrus-melon**
This updates the code to redefine the following terminology:
- "Arm" (previously "scoring system"): the combination of the pivot and elevator, responsible for moving the intake
- "Pivot" (previously "arm"): the rotating joint on the robot

- The previous `Arm` and `Elevator` subsystems are now non-subsystem `Pivot` and `Elevator` static nested classes of the `Arm` subsystem.
- Renames `ScoringPosition` to `ArmPose`
- Adds shortcut methods to the new `Arm` that return an `ArmPose` representing the current or desired state.
- Modifies docs and variable names for consistency and correctness

Closes #61

### CODE REVIEW

File: src/main/java/frc/robot/commands/arm/ArmTowardsPoseCommand.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  /**
+   * Creates a command that moves the {@link Arm} towards a given pose.
+   *
+   * @param arm The {@link Arm} to control
+   * @param pose A supplier for the pose to move towards
+   */
+  public ArmTowardsPoseCommand(Arm arm, Supplier<ArmPose> pose) {
+    this.addCommands(new RunElevatorToExtensionCommand(arm, () -> pose.get().getExtension()),
+        new RotatePivotToAngleCommand(arm, () -> pose.get().getAngle().getDegrees()));
+    addRequirements(arm);
```
Comment (kchenturtles) [inline src/main/java/frc/robot/commands/arm/ArmTowardsPoseCommand.java:39]: Parallel commands can't require the same subsystem
Maybe make the two commands not require the arm subsystem?

File: src/main/java/frc/robot/subsystems/Arm.java [hunk 2]
```diff
@@ -1,154 +1,307 @@
...
-   * 
-   * @param speed The desired rotation speed
+   * A sub-subsystem that controls the rotating pivot on the robot.
    */
-  public void setRotationSpeed(double speed) {
-    if (getAngle() < ArmConstants.MIN_ROTATION_DEG && motors.get() < 0
-        || getAngle() > ArmConstants.MAX_ROTATION_DEG && motors.get() > 0) {
-      motors.set(0);
-      return;
+  public static class Pivot {
```
Comment (kchenturtles) [inline src/main/java/frc/robot/subsystems/Arm.java]: Pivot shouldn't need to be static

File: src/main/java/frc/robot/subsystems/Arm.java [hunk 3]
```diff
@@ -1,154 +1,307 @@
...
   }
 
   /**
-   * Feeds the PID input to the motors.
+   * A sub-subsystem that controls the extending elevator on the robot.
    */
-  public void feedPID() {
-    setRotationSpeed(pidController.calculate(getAngle()));
-  }
+  public static class Elevator {
```
Comment (kchenturtles) [inline src/main/java/frc/robot/subsystems/Arm.java]: Elevator also doesn't need to be static

**Review by kchenturtles — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Arm.java (hunk 2 above)
Comment (citrus-melon) [reply src/main/java/frc/robot/subsystems/Arm.java]: It does, `static` on a class is to allow a nested class to be instantiated independent of any instance of its parent.

Comment (kchenturtles) [reply src/main/java/frc/robot/subsystems/Arm.java]: can we just avoid instantiating separately?

**Review by kchenturtles — APPROVED** [review]

**Pull request merged**